### PR TITLE
Unify i18n logic in the various Craft apps

### PR DESCRIPTION
**Pull request by Hamms**
# Description

Follow-up to https://github.com/code-dot-org/code-dot-org/pull/32136

Right now, all the Craft apps share the same i18n file, but each one also references that file independently and reimplements a bunch of shared logic around i18n. This change brings all shared i18n code up one level to be used by all Craft apps equally.

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-916)

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by joshlory** [comment]
To make sure I understand — this de-dups the localization across the four Craft variants, and merges all the strings into one file? Did you programmatically build the superset or create it by hand?

File: apps/test/unit/craft/i18nTest.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {assert} from '../../util/deprecatedChai';
```
Comment (joshlory) [inline apps/test/unit/craft/i18nTest.js]: Any reason not to use `reconfiguredChai`?

Comment (Hamms) [reply apps/test/unit/craft/i18nTest.js]: no idea! I was just copying what's in the existing `craftTest`

**Comment by Hamms** [comment]
> this de-dups the localization across the four Craft variants, and merges all the strings into one file?

precisely!

> Did you programmatically build the superset or create it by hand?

Bit of both; I copied all objects into one, sorted the keys by name, and deleted duplicates.

**Comment by joshlory** [comment]
Ah, cool. Looks like the end result was exactly the same as what the "agent" version had:

image

**Review by joshlory — APPROVED** [review]

**Review by davidsbailey — APPROVED** [review]
LGTM! Sorry for missing this the first time around.

> Ah, cool. Looks like the end result was exactly the same as what the "agent" version had:
> 
> image

This seems strange at first, but can probably be explained if each subsequent minecraft copied the previous one's locale file and then added to it.

**Pull request merged**